scripts/us_census/api_utils/url_list_compiler.py

```diff
@@ -0,0 +1,285 @@
+                                                 f"{s_dict['str']}:*",
+                                                 output_path, force_fetch_data))
+                else:
+                    print('Warning:', s_level, 'not available for year', year)
+    ret_list = sync_status_list([], ret_list)
+    return ret_list
+
+
+def get_variables_url_list(dataset: str,
+                           table_id,
+                           q_variable,
+                           variables_year_dict,
+                           output_path,
+                           api_key,
+                           s_level_list='all',
+                           force_fetch_config=False,
+                           force_fetch_data=False):
+    # TODO implement the method
+    pass
+    # table_id = table_id.upper()
+    # ret_list = []
+
+    # geo_config = get_summary_level_config(dataset, q_variable, api_key, force_fetch_config)
+    # # get list of all s levels
+    # if s_level_list == 'all':
+    #     s_level_list = []
+    #     for year, year_dict in geo_config.items():
+    #         for s_level in year_dict['summary_levels']:
+    #             if s_level not in s_level_list:
+    #                 s_level_list.append(s_level)
+
+    # year_list = []
+    # for year in variables_year_dict:
+    #     year_list.append(year)
+
+    # for year in variables_year_dict:
+    #     # limited to 50 variables including NAME
+    #     n = 49
+    #     variables_chunked = [variables_year_dict[year][i:i + n] for i in range(0, len(variables_year_dict[year]), n)]
+    #     logging.info('variable list divided into %d chunks for year %d', len(variables_chunked), year)
+    # for geo_id in geo_url_map:
+    #     geo_str = geo_url_map[geo_id]['urlStr']
+    #     if geo_url_map[geo_id]['needsStateID']:
+    #         for state_id in states_by_year[year]:
+    #             geo_str_state = geo_str + state_id
+    #             for i, cur_vars in enumerate(variables_chunked):
+    #                 variable_list_str = ','.join(cur_vars)
+    #                 temp_dict = {}
+    #                 temp_dict['url'] = get_url_variables(dataset, year, 'NAME,' + variable_list_str, geo_str_state)
+    #                 temp_dict['store_path'] = get_file_name_variables(output_path, table_id, year, i, geo_str_state)
+    #                 temp_dict['status'] = 'pending'
+    #                 ret_list.append(temp_dict)
+    #     else:
+    #         for i, cur_vars in enumerate(variables_chunked):
+    #             variable_list_str = ','.join(cur_vars)
+    #             temp_dict = {}
+    #             temp_dict['url'] = get_url_variables(dataset, year, 'NAME,' + variable_list_str, geo_str)
+    #             temp_dict['store_path'] = get_file_name_variables(output_path, table_id, year, i, geo_str)
+    #             temp_dict['status'] = 'pending'
+    #             ret_list.append(temp_dict)
+    # return ret_list
```

Comment:
Remove this block if it is not needed?

Reply:
Have kept the block as it acts as a basic code which needs minor modification. The feature is not useful currently but would be helpful in future and would need minor modifications only. It was hard to implement and test it within the current timeframe.